packer/scripts/Artifacts/MsSql/Structure/Ods/2050-CreateTableValueFunctions.sql

```diff
@@ -0,0 +1,574 @@
+        AND gspa.StudentUSI = fspa.StudentUSI
+    INNER JOIN edfi.StudentSchoolFoodServiceProgramAssociationSchoolFoodServiceProgramService fspas WITH (NOLOCK) ON
+        fspa.BeginDate = fspas.BeginDate
+        AND fspa.ProgramEducationOrganizationId = fspas.ProgramEducationOrganizationId
+        AND fspa.ProgramName = fspas.ProgramName
+        AND fspa.StudentUSI = fspas.StudentUSI
+    INNER JOIN edfi.Descriptor dFSP WITH (NOLOCK) ON
+        fspas.SchoolFoodServiceProgramServiceDescriptorId = dFSP.DescriptorId
+    INNER JOIN (
+            SELECT StudentUSI, ProgramEducationOrganizationId, MAX(BeginDate) as BeginDate
+            FROM edfi.GeneralStudentProgramAssociation WITH (NOLOCK)
+            WHERE
+                ProgramName = 'School Food Service'
+                AND BeginDate <= @searchDate
+                AND (EndDate IS NULL OR EndDate >= @searchDate)
+            GROUP BY
+                StudentUSI,
+                ProgramEducationOrganizationId
+        ) mbd ON
+        gspa.StudentUSI = mbd.StudentUSI
+        AND gspa.ProgramEducationOrganizationId = mbd.ProgramEducationOrganizationId
+        AND gspa.BeginDate = mbd.BeginDate;
+
+    RETURN;
+END;
+GO
+
+CREATE OR ALTER FUNCTION reporting.GetEnglishLearners
+(
+    @searchDate DATE
+)
+RETURNS @ResultTable TABLE
+(
+    StudentUSI INT,
+    ProgramEducationOrganizationId INT,
+    EnglishLearnerParticipation BIT,
+    RedesignatedEnglishFluent BIT,
+    INDEX idx_EnglishLearners CLUSTERED (ProgramEducationOrganizationId, StudentUSI)
+)
+AS
+BEGIN
+    INSERT INTO @ResultTable (StudentUSI, ProgramEducationOrganizationId, EnglishLearnerParticipation, RedesignatedEnglishFluent)
+    SELECT DISTINCT
+        gspa.StudentUSI,
+        gspa.ProgramEducationOrganizationId,
+        slipa.EnglishLearnerParticipation,
+        slipae.RedesignatedEnglishFluent
+    FROM edfi.GeneralStudentProgramAssociation gspa WITH (NOLOCK)
+    INNER JOIN edfi.EducationOrganization eo ON
+        gspa.ProgramEducationOrganizationId = eo.EducationOrganizationId
+    INNER JOIN edfi.StudentLanguageInstructionProgramAssociation slipa WITH (NOLOCK) ON
+        gspa.BeginDate = slipa.BeginDate
+        AND gspa.ProgramEducationOrganizationId = slipa.ProgramEducationOrganizationId
+        AND gspa.ProgramName = slipa.ProgramName
+        AND gspa.StudentUSI = slipa.StudentUSI
+    INNER JOIN sk.StudentLanguageInstructionProgramAssociationExtension slipae WITH (NOLOCK) ON
+        slipa.BeginDate = slipae.BeginDate
+        AND slipa.ProgramEducationOrganizationId = slipae.ProgramEducationOrganizationId
+        AND slipa.ProgramName = slipae.ProgramName
+        AND slipa.StudentUSI = slipae.StudentUSI
+    INNER JOIN
+        (
+            SELECT StudentUSI, ProgramEducationOrganizationId, MAX(BeginDate) as BeginDate
+            FROM edfi.GeneralStudentProgramAssociation WITH (NOLOCK)
+            WHERE
+                ProgramName = 'English Learners'
```

Comment:
@jamessmckay change this to 'English as a Second Language (ESL)'

Reply:
done